packages/fxa-content-server/app/scripts/models/attached-clients.js

```diff
@@ -9,61 +9,43 @@
  */
 import Backbone from 'backbone';
 import Constants from '../lib/constants';
-import Device from './device';
-import OAuthApp from './oauth-app';
-import WebSession from './web-session';
+import AttachedClient from './attached-client';
 
 var AttachedClients = Backbone.Collection.extend({
-  model: function(attrs, options) {
-    if (attrs.clientType === Constants.CLIENT_TYPE_DEVICE) {
-      return new Device(attrs, options);
-    } else if (attrs.clientType === Constants.CLIENT_TYPE_OAUTH_APP) {
-      return new OAuthApp(attrs, options);
-    } else if (attrs.clientType === Constants.CLIENT_TYPE_WEB_SESSION) {
-      return new WebSession(attrs, options);
-    }
+  model (attrs, options) {
+    return new AttachedClient(attrs, options);
   },
 
-  fetchClients (clientTypes = {}, user) {
-    var account = user.getSignedInAccount();
-    var fetchItems = [];
-
-    if (clientTypes.devices) {
-      fetchItems.push(user.fetchAccountDevices(account));
-    }
-
-    if (clientTypes.oAuthApps) {
-      fetchItems.push(user.fetchAccountOAuthApps(account));
-    }
-
-    if (clientTypes.sessions) {
-      fetchItems.push(user.fetchAccountSessions(account));
-    }
+  modelId (attrs) {
+    // Make a synthetic `id` attribute because attached clients don't have
+    // a single stable unique identifier. It's only for our internal use.
+    return `${attrs.clientId || ''}:${attrs.deviceId || ''}:${attrs.refreshTokenId || ''}:${attrs.sessionTokenId || ''}`;
+  },
```

Comment:
This is kinda gross, happy to take suggestions for an alternative way of doing it.